pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -166,13 +166,20 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		log.V(1).Info("skipping refresh", "rv", getResourceVersion(externalSecret))
 		return ctrl.Result{RequeueAfter: refreshInt}, nil
 	}
+	if externalSecret.Status.SyncedResourceVersion != "" && externalSecret.Spec.Target.Immutable {
+		return ctrl.Result{
+			RequeueAfter: 0,
+			Requeue:      false,
+		}, nil
+	}
```

Comment:
We should check externalSecret.Status.Conditions here to check if it has ConditionReasonSecretSynced, not the SyncedResourceVersion, I think.

Also better to add this logic to the shouldRefresh function, so it skips with new checks instead of a new if on the Reconcile.

Reply:
Since !shouldRefresh does not return the result with Requeue: false, I created a shouldRenconcile function to do so.